src/commands/cli/artifacts/compare.ts

```diff
@@ -0,0 +1,592 @@
+          this.log(chalk.red('  ❌ Flag Removals:'), cmd.flagRemovals);
+        }
+        if (cmd.aliasAdditions.length) this.log(chalk.dim('  Alias Additions:'), cmd.aliasAdditions);
+        if (cmd.aliasRemovals.length) {
+          summary[plugin].push(`${cmd.command} alias removals: ${cmd.aliasRemovals.join(', ')}`);
+          this.log(chalk.red('  ❌ Alias Removals:'), cmd.aliasRemovals);
+        }
+      }
+      this.log();
+      this.log(chalk.underline.cyan('Schema Changes'));
+      const humanReadableChanges = SchemaComparator.makeReadable(
+        result.current.schemas,
+        result.previous.schemas,
+        result.schemaChanges
+      );
+      if (Object.keys(humanReadableChanges).length === 0) {
+        this.log(chalk.dim('No changes have been detected.'));
+      }
+
+      for (const [commandId, readableChanges] of Object.entries(humanReadableChanges)) {
+        this.log();
+        this.log(commandId);
+        for (const change of readableChanges) {
+          if (change.startsWith('❌')) summary[plugin].push(change.replace('❌', ''));
+          this.log(`  ${change}`);
+        }
+      }
+      this.log();
+    }
+
+    return summary;
+  }
+
+  private showRemovedPlugins(): string[] {
+    const removedPlugins = Object.keys(this.previousPlugins).filter((p) => !this.currentPlugins[p]);
+    if (removedPlugins.length > 0) {
+      this.styledHeader(chalk.red('Removed Plugins'));
+      for (const plugin of removedPlugins) {
+        this.log(plugin);
+      }
+    }
+    return removedPlugins;
+  }
+
+  private showAddedPlugins(): string[] {
+    const addedPlugins = Object.keys(this.currentPlugins).filter((p) => !this.previousPlugins[p]);
+    if (addedPlugins.length > 0) {
+      this.styledHeader(chalk.green('Added Plugins'));
+      for (const plugin of addedPlugins) {
+        this.log(plugin);
+      }
+    }
+    return addedPlugins;
+  }
+
+  private resolveVersions(): void {
+    this.current = this.flags.current || this.packageJson.version;
+    this.previous = this.flags.previous ?? this.versions[this.versions.indexOf(this.current) + 1];
+    this.log('Current Version:', this.current);
+    this.log('Previous Version:', this.previous);
+    if (this.flags.current && !this.versions.includes(this.flags.current)) {
+      throw messages.createError('error.VersionNotFound', [this.flags.current]);
+    }
+    if (this.flags.previous && !this.versions.includes(this.flags.previous)) {
+      throw messages.createError('error.VersionNotFound', [this.flags.previous]);
+    }
+  }
+
+  private async getCurrentPlugins(): Promise<Record<string, string>> {
+    return this.flags.current ? this.getPluginsForVersion(this.current) : this.filterPlugins(this.packageJson);
+  }
+
+  private async getPluginsForVersion(version: string): Promise<Record<string, string>> {
+    if (!this.versions.includes(version)) {
+      throw messages.createError('error.VersionNotFound', [version]);
```

Comment:
Is this needed here since we have already checked that the version exists in `resolveVersions()`?

Reply:
Oops - just a vestige of a pervious implementation. I'll remove it.